lib/Agrammon/Web/Routes.pm6

```diff
@@ -94,11 +94,13 @@ sub dataset-routes(Agrammon::Web::Service $ws) {
             }
         }
 
-        # test
+        # working
         post -> LoggedIn $user, 'rename_dataset' {
-            request-body -> (:$old!, :$new!) {
-                my $data = $ws.rename-dataset($user, $old, $new);
-                content 'application/json', $data;
+            request-body -> (:oldName($old-name)!, :newName($new-name)!) {
+                my %ret = $ws.rename-dataset($user, $old-name, $new-name);
+                content 'application/json', %(
+                    :oldName(%ret<old>), :newName(%ret<new>)
```

Comment:
Hm, why is it returning the same thing that was sent in? Can it ever differ?

Reply:
The renaming could fail (because dataset names must be unique). I should probably handle that explicitly.
Would you rather return sucess/failure or so?

Reply:
Yes. I mean, really I'd rather see us use HTTP status codes to indicate failure rather than a 200 OK followed by a body telling us it's not OK. But I'm sympathetic to wanting to at least get things working with the current frontend before such changes.

Reply:
Probably 409 Conflict is the appropriate choice here (and Cro even provides a `conflict` sub for producing that response).

Reply:
I'll give it a try. As I am already making (small) changes to the frontend code, I think it should be ok to handle that properly.